### DISCUSSION TITLE
cfengine install (bottle) permissions are wrong, install from source are right

### DISCUSSION THREAD

**Discussion by craigcomstock** [Homebrew/discussions#4283] [Tap maintenance and Homebrew development] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.4-86-g7c15dce
ORIGIN: https://github.com/Homebrew/brew
HEAD: 7c15dce285900c85476d445dcbc01418e61be37c
Last commit: 6 hours ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 747f81671033408c3557c337cb576ffaeea7eb1c
Core tap last commit: 28 minutes ago
Core tap branch: master
Core tap JSON: 28 Feb 20:20 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: ed
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_NO_INSTALL_FROM_API: set
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit icelake
Clang: 14.0.0 build 1400
Git: 2.39.2 => /usr/local/bin/git
Curl: 7.86.0 => /usr/bin/curl
macOS: 13.2.1-x86_64
CLT: 14.2.0.0.1.1668646533
Xcode: N/A
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Homebrew's "sbin" was not found in your PATH but you have installed
formulae that put executables in /usr/local/sbin.
Consider setting your PATH for example like so:
  echo 'export PATH="/usr/local/sbin:$PATH"' >> ~/.zshrc
```

### Description of issue

If I do `brew install cfengine` I see that /usr/local/var/cfengine/ppkeys directory is owned $USER:admin and permissions are 0755. This is incorrect and will cause errors when someone tries to, for example, self bootstrap with `sudo cf-agent -IB localhost`.

If I do `brew install -s cfengine` the directories there are as I expect based on `make install` in the formula install [function](https://github.com/Homebrew/homebrew-core/blob/master/Formula/cfengine.rb#L52 ) and our [Makefile.am](https://github.com/cfengine/core/blob/master/Makefile.am#L91) 

I grabbed the bottle that is downloaded, un-tarred it and found that it had incorrect permissions of 0755.

Is this expected? Is there some way I can ensure proper permissions in the bottle via a change in the formula? Or do I need to add some post-install hook to change permissions to what we need them to be?

**Comment by gromgit** [comment, marked as answer]
https://github.com/Homebrew/homebrew-core/pull/124514

**Reply by gromgit** [reply]
The fix is in. `brew update`, `brew reinstall cfengine`, then see if the new permissions work for you.

**Comment by craigcomstock** [comment]
Tested and works perfectly. Thanks for the fix and the education on how to fix (the PR). Cheers!
